### PR TITLE
Cover resumption with unit tests

### PR DESCRIPTION
**Pull request by Kozoriz**
- SQLDatabase class divided  to interface and impl.
- Created mock for SQLDatabase 
- Added tests for resumptionDataDB, resumptionDataJson, resumptionCtrl classes

### CODE REVIEW

**Comment by Kozoriz** [comment]
please review : @anosach-luxoft , @dev-gh , @VProdanov , @LevchenkoS, @okozlovlux

File: src/components/application_manager/test/resumption/resumption_data_db_test.cc [hunk 1]
```diff
@@ -66,8 +76,9 @@ const std::string kPath =
 }
 class ResumptionDataDBTest : public ResumptionDataTest {
  protected:
-  void SetUp() OVERRIDE {
-    app_mock = utils::MakeShared<NiceMock<MockApplication> >();
+  virtual void SetUp() {
```
Comment (dev-gh) [inline src/components/application_manager/test/resumption/resumption_data_db_test.cc]: @Kozoriz 
Inconsistent with other code.

File: src/components/application_manager/test/resumption/resumption_data_db_test.cc [hunk 2]
```diff
@@ -77,20 +88,18 @@ class ResumptionDataDBTest : public ResumptionDataTest {
     ign_off_count_ = 0;
     grammar_id_ = 16;
   }
-  void TearDown() OVERRIDE {
-    utils::dbms::SQLQuery query(test_db());
+  virtual void TearDown() {
```
Comment (dev-gh) [inline src/components/application_manager/test/resumption/resumption_data_db_test.cc]: @Kozoriz 
Inconsistent with other code

File: src/components/application_manager/test/resumption/resumption_data_json_test.cc [hunk 3]
```diff
@@ -1,5 +1,5 @@
 /*
- * Copyright (c) 2016, Ford Motor Company
+ * Copyright (c) 2015, Ford Motor Company
```
Comment (dev-gh) [inline src/components/application_manager/test/resumption/resumption_data_json_test.cc]: @Kozoriz 
Seems changes have been rolled back

**Comment by dev-gh** [comment]
Reviewed

**Comment by LevchenkoS** [comment]
@Kozoriz, reviewed.

File: src/components/application_manager/src/resumption/resumption_data_db.cc [hunk 4]
```diff
@@ -401,32 +406,6 @@ void ResumptionDataDB::GetDataForLoadResumeData(
   SelectDataForLoadResumeData(saved_data);
```
Comment (VProdanov) [inline src/components/application_manager/src/resumption/resumption_data_db.cc]: @Kozoriz can you please check 
bool ResumptionDataDB::RemoveApplicationFromSaved.. a few lines above.

File: src/components/utils/src/sqlite_wrapper/sql_database_impl.cc [hunk 5]
```diff
@@ -41,36 +41,32 @@ const char* kInMemory = ":memory:";
...
 
-SQLDatabase::~SQLDatabase() {
+SQLDatabaseImpl::~SQLDatabaseImpl() {
   Close();
 }
 
-bool SQLDatabase::Open() {
+bool SQLDatabaseImpl::Open() {
   sync_primitives::AutoLock auto_lock(conn_lock_);
   if (conn_)
```
Comment (VProdanov) [inline src/components/utils/src/sqlite_wrapper/sql_database_impl.cc]: @Kozoriz curly brackets are missing.

**Comment by VProdanov** [comment]
Reviewed

File: src/components/application_manager/test/resumption/resume_ctrl_test.cc [hunk 6]
```diff
@@ -695,16 +774,33 @@ TEST_F(ResumeCtrlTest, GetHMIApplicationID) {
 }
 
 TEST_F(ResumeCtrlTest, IsApplicationSaved) {
-  std::string policy_app_id = "policy_app_id";
-  std::string device_id = "device_id";
+  const std::string policy_app_id = "policy_app_id";
+  const std::string device_id = "device_id";
```
Comment (okozlovlux) [inline src/components/application_manager/test/resumption/resume_ctrl_test.cc]: @Kozoriz please, rename consts according to coding style

File: src/components/application_manager/test/resumption/resume_ctrl_test.cc [hunk 7]
```diff
@@ -695,16 +774,33 @@ TEST_F(ResumeCtrlTest, GetHMIApplicationID) {
...
+  uint32_t timeout = 10u;
+  EXPECT_CALL(mock_application_manager_settings_,
+              app_resumption_save_persistent_data_timeout())
+      .WillOnce(ReturnRef(timeout));
+  res_ctrl->OnAppRegistrationEnd();
+}
+
 TEST_F(ResumeCtrlTest, CheckPersistenceFiles_WithoutCommandAndChoiceSets) {
-  uint32_t ign_off_count = 0;
+  const uint32_t ign_off_count = 0;
```
Comment (okozlovlux) [inline src/components/application_manager/test/resumption/resume_ctrl_test.cc]: @Kozoriz please, check coding style with const names

File: src/components/application_manager/src/resumption/resumption_data_db.cc (hunk 4 above)
Comment (Kozoriz) [inline src/components/application_manager/src/resumption/resumption_data_db.cc]: @VProdanov , what I must check ?

Comment (VProdanov) [inline src/components/application_manager/src/resumption/resumption_data_db.cc]: @Kozoriz its just that the function return with resuly on one place and with false on the other.. I was thinking if the declaration is made before ExistenceCheck.. and if we fail to return result ..

File: src/components/application_manager/test/resumption/resumption_data_db_test.cc (hunk 1 above)
Comment (Kozoriz) [inline src/components/application_manager/test/resumption/resumption_data_db_test.cc]: @dev-gh , done 
	6f5b1dc

File: src/components/application_manager/test/resumption/resumption_data_db_test.cc (hunk 2 above)
Comment (Kozoriz) [inline src/components/application_manager/test/resumption/resumption_data_db_test.cc]: @dev-gh , done 
	6f5b1dc

**Comment by dev-gh** [comment]
Reviewed

File: src/components/application_manager/test/resumption/resume_ctrl_test.cc (hunk 7 above)
Comment (Kozoriz) [inline src/components/application_manager/test/resumption/resume_ctrl_test.cc]: @okozlovlux , done 6f5b1dc

File: src/components/application_manager/test/resumption/resumption_data_json_test.cc (hunk 3 above)
Comment (Kozoriz) [inline src/components/application_manager/test/resumption/resumption_data_json_test.cc]: @dev-gh , done 6f5b1dc

**Comment by okozlovlux** [comment]
@Kozoriz reviewed

File: src/components/application_manager/src/resumption/resumption_data_db.cc (hunk 4 above)
Comment (Kozoriz) [inline src/components/application_manager/src/resumption/resumption_data_db.cc]: @VProdanov done d2d60d0

File: src/components/utils/src/sqlite_wrapper/sql_database_impl.cc (hunk 5 above)
Comment (Kozoriz) [inline src/components/utils/src/sqlite_wrapper/sql_database_impl.cc]: @VProdanov , done d2d60d0

**Comment by anosach-luxoft** [comment]
Reviewed

File: src/components/utils/include/utils/sql_wrapper.h [hunk 8]
```diff
@@ -30,18 +30,18 @@
...
  */
 
-#ifndef SRC_COMPONENTS_POLICY_INCLUDE_POLICY_SQL_WRAPPER_H_
-#define SRC_COMPONENTS_POLICY_INCLUDE_POLICY_SQL_WRAPPER_H_
+#ifndef SRC_COMPONENTS_UTILS_INCLUDE_UTILS_SQL_WRAPPER_H_
+#define SRC_COMPONENTS_UTILS_INCLUDE_UTILS_SQL_WRAPPER_H_
 
 #if __QNX__
-#include "qdb_wrapper/sql_database.h"
+#include "utils/sqlite_wrapper/sql_database_impl.h"
```
Comment (AGaliuzov) [inline src/components/utils/include/utils/sql_wrapper.h]: @Kozoriz Have you removed `qdb_wrapper` directory?

Comment (Kozoriz) [inline src/components/utils/include/utils/sql_wrapper.h]: @AGaliuzov , no. my mistake, corrected in c381b0f

**Pull request merged**